samcli/commands/sync/command.py

```diff
@@ -142,6 +143,7 @@
 @track_template_warnings([CodeDeployWarning.__name__, CodeDeployConditionWarning.__name__])
 @check_newer_version
 @print_cmdline_args
+@unsupported_command_cdk(alternative_command=None)
```

Comment:
If you make `alternative_command=None` as the default, you don't have to pass in the command unless you need to for specific commands.